# **Análise exploratória de dados**

Este projeto possui uma análise aprofundada dos dados selecionados para identificar padrões, tendências e insights relevantes.

**Descrição:** Escolha um conjunto de dados de interesse e explore suas características. Limpe e prepare os dados, em seguida, utilize técnicas de visualização, estatística descritiva e exploração interativa para identificar insights valiosos. Destaque padrões, tendências ou anomalias notáveis presentes nos dados.

# **Sobre o dataset**

O dataset deste projeto, abrange informações sobre os salários de diversos cargos dentro do campo da Ciência de Dados. Ele fornece uma visão detalhada das remunerações em diferentes funções, como cientista de dados, engenheiro de machine learning, analista de dados, entre outros.

**Link para o dataset:** https://www.kaggle.com/datasets/arnabchaki/data-science-salaries-2023

#**Limpeza e preparação dos dados**

Eu gosto de começar analisando o dataset original em busca de dados que sejam interessantes.

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('ds_salaries.csv')

df.head(20)

work_year experience_level employment_type                 job_title  \
0        2023               SE              FT  Principal Data Scientist   
1        2023               MI              CT               ML Engineer   
2        2023               MI              CT               ML Engineer   
3        2023               SE              FT            Data Scientist   
4        2023               SE              FT            Data Scientist   
5        2023               SE              FT         Applied Scientist   
6        2023               SE              FT         Applied Scientist   
7        2023               SE              FT            Data Scientist   
8        2023               SE              FT            Data Scientist   
9        2023               SE              FT            Data Scientist   
10       2023               SE              FT            Data Scientist   
11       2023               SE              FT              Data Analyst   
12       2023               SE              FT              Data Analyst   
13       2023               EN              FT         Applied Scientist   
14       2023               EN              FT         Applied Scientist   
15       2023               SE              FT              Data Modeler   
16       2023               SE              FT              Data Modeler   
17       2023               SE              FT            Data Scientist   
18       2023               SE              FT            Data Scientist   
19       2023               MI              FT              Data Analyst   

    salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0    80000             EUR          85847                 ES           100   
1    30000             USD          30000                 US           100   
2    25500             USD          25500                 US           100   
3   175000             USD         175000                 CA           100   
4   120000             USD         120000                 CA           100   
5   222200             USD         222200                 US             0   
6   136000             USD         136000                 US             0   
7   219000             USD         219000                 CA             0   
8   141000             USD         141000                 CA             0   
9   147100             USD         147100                 US             0   
10   90700             USD          90700                 US             0   
11  130000             USD         130000                 US           100   
12  100000             USD         100000                 US           100   
13  213660             USD         213660                 US             0   
14  130760             USD         130760                 US             0   
15  147100             USD         147100                 US             0   
16   90700             USD          90700                 US             0   
17  170000             USD         170000                 US             0   
18  150000             USD         150000                 US             0   
19  150000             USD         150000                 US           100   

   company_location company_size  
0                ES            L  
1                US            S  
2                US            S  
3                CA            M  
4                CA            M  
5                US            L  
6                US            L  
7                CA            M  
8                CA            M  
9                US            M  
10               US            M  
11               US            M  
12               US            M  
13               US            L  
14               US            L  
15               US            M  
16               US            M  
17               US            M  
18               US            M  
19               US            M

In [15]:
colunas = df[['work_year', 'experience_level', 'job_title', 'salary_in_usd', 'employee_residence', 'company_location', 'company_size']]

colunas.head(20)

work_year experience_level                 job_title  salary_in_usd  \
0        2023               SE  Principal Data Scientist          85847   
1        2023               MI               ML Engineer          30000   
2        2023               MI               ML Engineer          25500   
3        2023               SE            Data Scientist         175000   
4        2023               SE            Data Scientist         120000   
5        2023               SE         Applied Scientist         222200   
6        2023               SE         Applied Scientist         136000   
7        2023               SE            Data Scientist         219000   
8        2023               SE            Data Scientist         141000   
9        2023               SE            Data Scientist         147100   
10       2023               SE            Data Scientist          90700   
11       2023               SE              Data Analyst         130000   
12       2023               SE              Data Analyst         100000   
13       2023               EN         Applied Scientist         213660   
14       2023               EN         Applied Scientist         130760   
15       2023               SE              Data Modeler         147100   
16       2023               SE              Data Modeler          90700   
17       2023               SE            Data Scientist         170000   
18       2023               SE            Data Scientist         150000   
19       2023               MI              Data Analyst         150000   

   employee_residence company_location company_size  
0                  ES               ES            L  
1                  US               US            S  
2                  US               US            S  
3                  CA               CA            M  
4                  CA               CA            M  
5                  US               US            L  
6                  US               US            L  
7                  CA               CA            M  
8                  CA               CA            M  
9                  US               US            M  
10                 US               US            M  
11                 US               US            M  
12                 US               US            M  
13                 US               US            L  
14                 US               US            L  
15                 US               US            M  
16                 US               US            M  
17                 US               US            M  
18                 US               US            M  
19                 US               US            M

Escolhi algumas colunas que chamaram a minha atenção.

O dataset possui alguns cargos dentro da área de análise de dados. Decidi trabalhar apenas com o cargo de 'Data Analyst'.

In [50]:
data_analyst = colunas[colunas['job_title'] == 'Data Analyst']

data_analyst.shape

(612, 7)

O dataset possui 612 linhas referentes ao cargo de Data Analyst.

In [54]:
anos = data_analyst.groupby('work_year')

anos_dict = {ano: grupo for ano, grupo in anos}

ano_20 = anos_dict.get(2020)
ano_21 = anos_dict.get(2021)
ano_22 = anos_dict.get(2022)
ano_23 = anos_dict.get(2023)

Após a limpeza e preparação do dataset, consegui ter uma ideia dos insights que desejo obter.

Temos dados de um período de 4 anos de salários para o cargo de Data Analyst.

Como primeiro objetivo, irei obter o salário médio para cada nível de experiência por ano.

In [ ]:
anos = [ano_20, ano_21, ano_22, ano_23]
experiencia = ['EN', 'MI', 'SE']

resultados = []

for ano in anos:
    resultados_ano = {}
    for nivel in experiencia:
        nivel_filtrado = ano.query("experience_level == @nivel")
        media_nivel = nivel_filtrado['salary_in_usd'].mean()
        resultados_ano[nivel] = media_nivel
    resultados.append(resultados_ano)

media_en_20 = pd.DataFrame({'EN Média 2020': [resultados[0]['EN']]})
media_mi_20 = pd.DataFrame({'MI Média 2020': [resultados[0]['MI']]})
media_se_20 = pd.DataFrame({'SE Mdia 2020': [resultados[0]['SE']]})

media_final_20 = pd.concat([media_en_20, media_mi_20, media_se_20], axis=1)

media_final_20.head()

O resultado final está errado. O próximo passo será resolver o problema que faz com que o valor final não seja correto.